### PR TITLE
Automatically create verified group when all users are verified

### PR DESCRIPTION
**Pull request by adzialocha**
This PR simplifies the UI by removing all options to manually create a verified group chat. The idea is to automatically do so if all selected group members are known to be verified.

If an unverified member is added to an verified group a dialogue will be displayed. 

https://github.com/deltachat/deltachat-desktop/assets/1822473/5dec1da8-8f00-4cdc-a9fa-002c6c2aba1d

This entails:

* [x] Remove "Create Verified Group" from "New Chat" screen
* [x] Remove "QR Invite Code" from "Create Group" screen
* [x] When only verified contacts are selected in "New Group" screen, automatically create a verified group - otherwise a normal group
* [x] Set new stock string `DC_STR_NEW_GROUP_SEND_FIRST_MESSAGE` to `chat_new_group_hint`
* [x] Remove verified/unverified filtering of contact lists when adding members
* [x] When tapping an unverified contact in a verified group, show an alert with the text `verified_contact_required_explain` and the answer `learn_more` / `qrscan_title` / `ok` with scan-qr - similar to what we're doing at https://github.com/deltachat/deltachat-desktop/pull/3328
* [x] Verified status changes automatically in "Add new member to group" dialogue (after QR was scanned)

Tracking issue: https://github.com/deltachat/deltachat-desktop/issues/3468

### CODE REVIEW

File: src/renderer/components/dialogs/ViewGroup.tsx [hunk 1]
```diff
@@ -110,22 +115,36 @@ export const useGroup = (chat: Type.FullChat) => {
...
+    const accountId = selectedAccountId()
+
+    // We can only add verified contacts to a protected group
+    if (chat.isProtected) {
+      const membersVerified = await areAllContactsVerified(
+        accountId,
+        newGroupMembers
+      )
+      if (!membersVerified) {
+        // @TODO: Show dialogue here
```
Comment (adzialocha) [inline src/renderer/components/dialogs/ViewGroup.tsx:134]: I'm not sure what the UX flow is for when a user tries to add an unverified contact to a verified group. I had a look at the [https://github.com/deltachat/deltachat-desktop/pull/3328](https://github.com/deltachat/deltachat-desktop/pull/3492/files#) flow. I assume we want to show the same dialogue (ProtectionBrokenDialog) then but a different QR code which allows the contact to scan it and join the group in a verified manner?

Comment (r10s) [reply src/renderer/components/dialogs/ViewGroup.tsx:134]: it is just not possibe to add unverified members, they must not even be selected

instead, an alert with "scan qr" and "learn more" is shown, but none of them will add unverified contacts to selected contacts

however, if the "qr code" scan allows to be opened while the add-member-dialog is still open (android/ios allow that), the list should be updated if verification-state changes

File: src/renderer/components/helpers/ChatMethods.ts [hunk 2]
```diff
@@ -338,11 +338,14 @@ export async function modifyGroup(
 ) {
   const accountId = selectedAccountId()
   log.debug('action - modify group', { chatId, name, image, members })
-  await BackendRemote.rpc.setChatName(accountId, chatId, name)
+
   const chat = await BackendRemote.rpc.getFullChatById(accountId, chatId)
   if (!chat) {
     throw new Error('chat is undefined, this should not happen')
   }
```
Comment (Simon-Laux) [inline src/renderer/components/helpers/ChatMethods.ts:345]: `await BackendRemote.rpc.getFullChatById(accountId, chatId)` will already throw in that case so this if statement can be removed, probably comes from the time of using the cffi api in desktop where there is not so good error handling (check for 0 as hint there was an error, then fish the error from the core events, sometimes not even that)

getFullChatById can not return `null` or `undefined`, this is it's rust function signature:
```rs
 async fn get_full_chat_by_id(&self, account_id: u32, chat_id: u32) -> Result<FullChat>
```

File: src/renderer/components/dialogs/ViewGroup.tsx (hunk 1 above)
Comment (adzialocha) [reply src/renderer/components/dialogs/ViewGroup.tsx]: > instead, an alert with "scan qr" and "learn more" is shown, but none of them will add unverified contacts to selected contacts

Super, thank you! This should be implemented now.

> however, if the "qr code" scan allows to be opened while the add-member-dialog is still open (android/ios allow that), the list should be updated if verification-state changes

I'm currently looking into that, not sure if it is part of this PR though as there are some edge cases to take into consideration:

1. Automatic update of newly added email address within the "add member" dialogue
2. Automatic update of existing addresses (for this I found the `ContactsChanged` event, but doesn't seem to be working, I'm still troubleshooting)

File: src/renderer/components/dialogs/ProtectionStatusDialog.tsx [hunk 3]
```diff
@@ -8,13 +9,16 @@ import {
...
-  Enabled: 'https://staging.delta.chat/733/en/help#verifiedchats',
-}
+import type { DialogProps } from './DialogController'
+
+const VERIFICATION_BROKEN_URL =
+  'https://staging.delta.chat/733/en/help#verificationbroken'
+const VERIFICATION_ENABLED_URL =
+  'https://staging.delta.chat/733/en/help#verifiedchats'
+const VERIFICATION_REQUIRED_URL =
+  'https://staging.delta.chat/746/en/help#howtoe2ee'
```
Comment (Simon-Laux) [inline src/renderer/components/dialogs/ProtectionStatusDialog.tsx:21]: maybe we should not duplicate it but move it instead to shared/constants file?

Comment (Simon-Laux) [reply src/renderer/components/dialogs/ProtectionStatusDialog.tsx:21]: ah sorry read it wrong nvm

**Review by Simon-Laux — APPROVED** [review]

**Pull request merged**